RELEASE/scripts/autoscend/iotms/mr2024.ash

```diff
@@ -649,16 +657,142 @@ boolean auto_thisClanPhotoBoothHasItems(boolean[item] its)
 }
 
 boolean auto_getClanPhotoBoothDefaultItems()
+{
+	return auto_getClanPhotoBoothDefaultItems(0);
+}
+
+boolean auto_getClanPhotoBoothDefaultItems(int reserve)
+{
+	if (!auto_haveClanPhotoBooth())
+	{
+		return false;
+	}
+	int n_claimable = auto_remainingClanPhotoBoothItems();
+	if (reserve < 0 || reserve > n_claimable)
+	{
+		auto_log_error("Bad value of items to reserve.");
+	}
+	
+	// Defult item logic:
+		// If fake arrow is better for -combat than any other hat we have, and we have no fireworks access to buy a popper, take fake arrow.
+		// If astronaut helmet is best cold red hat we have available and we'll mouthwash level, take fake arrow.
+		// If turbo or day count > 1, take off-stat sheriff items, matched to tower if scope available.
+		// Take monocle if best-in-slot for item (special case for unbrella)
+		// Take feather boa
+    
+	boolean[item] items_to_claim;
+	
+	int[item] hats_available = auto_getAllEquipabble($slot[hat]);
+	
+	// Check fake arrow through the head
+	item[int] hats_sorted_by_nc = auto_sortedByModifier(hats_available,$modifier[combat rate],false);
+	float get_combat(item it)
+	{
+		return numeric_modifier(it,$modifier[combat rate]);
+	}
+	boolean arrow_is_best_nc_hat = count(hats_sorted_by_nc)==0 || (get_combat($item[fake arrow-through-the-head]) < get_combat(hats_sorted_by_nc[0]));
+	if (arrow_is_best_nc_hat && !have_fireworks_shop()) // If we have fireworks we'll get porkpie-mounted popper
+	{
+		items_to_claim[$item[fake arrow-through-the-head]] = true;
+	}
+	
+	if (count(items_to_claim) == n_claimable)
+	{
+		return auto_getClanPhotoBoothItems(items_to_claim);
+	}
+	
+	// Check astronaut helmet
+	item[int] hats_sorted_by_cold_res = auto_sortedByModifier(hats_available,$modifier[cold resistance]);
+	float get_res(item it)
+	{
+		return numeric_modifier(it,$modifier[cold resistance]);
+	}
+	boolean astronaut_is_best_cold_hat = count(hats_sorted_by_cold_res)==0 || (get_res($item[astronaut helmet]) > get_res(hats_sorted_by_cold_res[0]));
+	if (auto_goingToMouthwashLevel() && astronaut_is_best_cold_hat)
+	{
+		items_to_claim[$item[astronaut helmet]] = true;
+	}
+
+	if (count(items_to_claim) == n_claimable)
+	{
+		return auto_getClanPhotoBoothItems(items_to_claim);
+	}
+	
+	// Check sheriff items (off-stat only, for tower)
+	// Don't do this d1 unless we're in turbo
+	if (my_daycount()>1 || auto_turbo())
+	{
+		stat challenge_stat = auto_getOffStatChallengeFromTelescope();
+		if (challenge_stat == $stat[muscle])
+		{
+			items_to_claim[$item[sheriff pistol]] = true;
+		}
+		else if (challenge_stat == $stat[mysticality])
+		{
+			items_to_claim[$item[sheriff badge]] = true;
+		}
+		else if (challenge_stat == $stat[moxie])
+		{
+			items_to_claim[$item[sheriff moustache]] = true;
+		}
+		else if (challenge_stat == $stat[none])
+		{
+			reserve++; // reserve a claim if no telescope
+		}
+	}
+	// To do: dual pistol support - items_to_claim only supports single entries per item
```

Comment:
you could reserve++ for pistol if you have DFSS.

You could even reserve instead of getting the items entirely; there isn't much use for them early.